backend/server/core/services/getTaskArchive.service.ts

```diff
@@ -11,30 +11,31 @@ import {StateItem} from "../enums/stateItem.enum";
 // type GetTaskArchive = (challengeId: number) => ArchiveItem[];
 
 export function getTaskArchive(
-    challengeId: number,
-    challenges: Challenge[]
+  challengeId: number,
+  challenges: Challenge[]
 ): ArchiveItem[] {
-    const challenge: Challenge = challenges.find(
-        (challenge) => challenge.id === challengeId
-    );
+  const challenge: Challenge = challenges.find(
+    (challenge) => challenge.id === challengeId
+  );
 
-    const timeBetween = new Date().getTime() - challenge.startDate.getTime();
-    const daysBetween = Math.floor(timeBetween / (1000 * 3600 * 24));
-    const passedTasks: Task[] = challenge.tasksOrder.slice(0, daysBetween);
+  const timeBetween = new Date().getTime() - challenge.startDate.getTime();
+  const daysBetween = Math.floor(timeBetween / (1000 * 3600 * 24));
+  const passedTasks: Task[] = challenge.tasksOrder.slice(0, daysBetween);
```

Comment:
Looks like you don't need lines 21 - 23 here.
This function works with finished challenges. For example, you can get April challenge in June. As the challenge is finished, some achievements and tasks in it may be finished, some no.
So you need:
1. find challenge by its `id`
2. get all achievements in this challenges
3. collect all tasks from all achievements in array and return this array